src/checkpoint/dispatch/reconstructor.h

```diff
@@ -51,7 +51,12 @@
 #include <tuple>
 #include <cstdlib>
 
-namespace checkpoint { namespace dispatch {
+namespace checkpoint {
+
+template <typename T>
+struct CheckpointReconstructor;
```

Comment:
I don't think you need to repeat the declaration here - it's already included.

Reply:
Other than this, looks good to me.

Reply:
Fixed